### DISCUSSION TITLE
Automation for Meta-Data and filenames

### DISCUSSION THREAD

**Discussion by HaukeGulich** [ciur/papermerge#308] [Ideas]
Hello

first of all thank you very much for this great project. I just bought a ScanSnap ix1500 and want to finally digitize my documents and came across your project. 

In principle perfect and exactly what I'm looking for. However, I came across two things that I would like to question again.

1) I find the meta-data concept super. But why can't the meta data be read in automatically? With the tags, this already works, although it is of course somewhat easier, because here only an "exists" or "does not exist" comes as a result. In the summer of last year there was already a discussion about it

https://github.com/ciur/papermerge/issues/30

And I wanted to ask if there is already news or a timeline for it? 

2) File names - basically the same issue. It would be great if the filenames could be automatically composed from meta data and/or tags. 

As an example, I would like to scan my pay stubs. The tag is already set correctly, but entering the meta data per hand (month, year and whether it's mine or my wife's) is a lot of work. I could realize the distinction between my wife and me via tags.

Maybe you can give us some feedback here.

Thanks for the effort and work and many greetings

**Comment by ciur** [comment]
Hi @HaukeGulich, thank you for bringing up this discussion.

Yes, you are absolutely right that the issue popped up since #30.
Good news, though -  almost all higher priority issues are fixed by now.

The automated metadata extraction is planned for version 2.1 and will be implemented very similarly to the concepts described in this [invoice2data project](https://github.com/invoice-x/invoice2data). Basically the automates will have an extra tab where you can upload an yml file describing the template of your document. For example:

```
metadata:
  amount: GrandTotal(\d+\.\d+)HKD
  date: DocumentDate:(\d{1,2}\/\d{1,2}\/\d{4})
  invoice_number: InvoiceNo.:(\w+)
options:
  remove_whitespace: true
  currency: HKD
  date_formats:
    - '%d/%m/%Y'
```
and metadata will be pulled out using regular expressions from OCR text and added to respective document.

Implementing this "automatic exaction" feature is not a big deal. The problem is that once your metadata will pile up you will want a convenient way to browse/search by metadata fields - which currently is not there (and not that snappy to implement).
Thus, those two features, automatic extraction and searching of by metadata fields, will be delivered together as they don't make sense one without other.

I will try my best to deliver those features by mid 2021.

**Reply by Nonaas** [reply]
Hey @ciur , are there any updates in regards to this? :-)
